WAVLTree/src/WAVLTree.java

```diff
@@ -110,10 +109,66 @@ public String max() {
 	 * respective keys, or an empty array if the tree is empty.
 	 */
 	public String[] infoToArray() {
-		String[] arr = new String[42]; // to be replaced by student code
+		String[] arr = new String[getRoot().size];
+		int[] count = new int[1];
+		if (!this.empty()) {
+			infoToArrayRec(this, arr, count);
+		}
 		return arr; // to be replaced by student code
 	}
 
+	/**
+	 * Used recursive inner function.
+	 * Needs to pass a tree, the array that we're populating with keys
+	 * and a coutner.
+	 * The counter is to validate that we're adding values in the right places.
+	 * Stopping conditions - if we reach an empty tree, we return without
+	 * changing the array (hope this works like Python or JS)
+	 * Otherwiase, we first run on the left side of the root,
+	 * then add the root to the array and move the counter + 1,
+	 * then move to the right.
+	 * Complexity - We need to go through all the nodes in the tree (n)
+	 * and we do a set number of O(1) operations for each, thus O(n)
+	 */
+	public String[] infoToArrayRec(WAVLTree tree, String[] array, int[] counter) {
+		if (!tree.empty()) {
+			return;
+		} else {
+			WAVLTree L = new WAVLTree(getRoot().getLeft());
+			infoToArrayRec(L, array, counter);
+			array[counter[0]] = getRoot().getValue();
+			counter[0] = counter[0] + 1;
+			WAVLTree R = new WAVLTree(getRoot().getLeft());
+			infoToArrayRec(R, array, counter);
+		}
+	}
+
+	/**
+	 * Used recursive inner function.
+	 * Needs to pass a tree, the array that we're populating with keys
+	 * and a coutner.
+	 * The counter is to validate that we're adding values in the right places.
+	 * Stopping conditions - if we reach an empty tree, we return without
+	 * changing the array (hope this works like Python or JS)
+	 * Otherwiase, we first run on the left side of the root,
+	 * then add the root to the array and move the counter + 1,
+	 * then move to the right.
+	 * Complexity - We need to go through all the nodes in the tree (n)
+	 * and we do a set number of O(1) operations for each, thus O(n)
+	 */
+	public String[] keysToArrayRec(WAVLTree tree, String[] array, int[] counter) {
```

Comment:
```suggestion
	private String[] keysToArrayRec(WAVLTree tree, String[] array, int[] counter) {
```

Reply:
All methods we only use internally should be private.

Reply:
👍